src/main/java/org/dependencytrack/tasks/CloneProjectTask.java

```diff
@@ -39,12 +44,29 @@ public void inform(final Event e) {
         if (e instanceof CloneProjectEvent) {
             final CloneProjectEvent event = (CloneProjectEvent)e;
             final CloneProjectRequest request = event.getRequest();
-            LOGGER.info("Cloning project: " + request.getProject());
-            try (QueryManager qm = new QueryManager()) {
+            final UUID chainIdentifier = ((CloneProjectEvent) e).getChainIdentifier();
+            final QueryManager qm = new QueryManager();
+            WorkflowState workflowState = qm.updateStartTimeIfWorkflowStateExists(chainIdentifier, PROJECT_CLONE);
+            if (workflowState == null) {
+                final var now = new Date();
+                workflowState = new WorkflowState();
+                workflowState.setStep(PROJECT_CLONE);
+                workflowState.setStatus(PENDING);
+                workflowState.setToken(chainIdentifier);
+                workflowState.setStartedAt(now);
+                workflowState.setUpdatedAt(now);
+                qm.getPersistenceManager().makePersistent(workflowState);
+            }
+            try {
+                LOGGER.info("Cloning project: " + request.getProject());
                 final Project project = qm.clone(UUID.fromString(request.getProject()),
                         request.getVersion(), request.includeTags(), request.includeProperties(),
                         request.includeComponents(), request.includeServices(), request.includeAuditHistory(), request.includeACL(), request.includePolicyViolations());
+                qm.updateWorkflowStateToComplete(workflowState);
                 LOGGER.info("Cloned project: " + request.getProject() + " to " + project.getUuid());
+            } catch (Exception ex) {
+                qm.updateWorkflowStateToFailed(workflowState, ex.getMessage());
```

Comment:
If we log the exception here, it is effectively handled and you don't need to re-throw it below. By re-throwing, it will be handled by the default uncaught exception handler, which we should avoid if possible.